src/common/functions.rs

```diff
@@ -12,6 +12,60 @@ use rand::{
 };
 use rust_decimal::prelude::*;
 use std::iter;
+use wasm_bindgen::prelude::*;
+use web_sys::js_sys::{Intl::NumberFormat, *};
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = navigator, js_name = language)]
+    static LANGUAGE: String;
+}
+
+#[wasm_bindgen]
+pub fn get_browser_locale() -> String {
+    LANGUAGE.clone()
+}
+
+fn format_number_helper(number: &str, max_significant_digits: Option<u32>) -> String {
+    let locale_array = Array::new();
+    locale_array.push(&JsValue::from_str(&get_browser_locale()));
+
+    let options = Object::new();
+    if let Some(digits) = max_significant_digits {
+        // Safely set the maximumSignificantDigits only if within a valid range:
+        // See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat
+        if (1..=21).contains(&digits) {
+            #[allow(unused_unsafe)]
+            unsafe {
+                Reflect::set(
+                    &options,
+                    &JsValue::from_str("maximumSignificantDigits"),
+                    &JsValue::from_f64(digits as f64),
+                )
+                .unwrap()
+            };
+        }
+    }
+
+    let num_format = NumberFormat::new(&locale_array, &options);
+    let format = num_format.format();
+    let this = JsValue::NULL; // `this` context for the function, not needed here
+
+    match format.call1(&this, &JsValue::from_str(number)) {
+        Ok(result) => result.as_string().unwrap_or_else(|| number.to_string()),
+        Err(_) => number.to_string(),
+    }
+}
+
+#[wasm_bindgen]
+pub fn format_number(number: String) -> String {
+    format_number_helper(&number, None) // No significant digits limit
+}
+
+#[wasm_bindgen]
+pub fn format_mina(number: String) -> String {
+    format_number_helper(&number, Some(9)) // Use 9 significant digits
```

Comment:
Is this really max significant digits, or is it number of digits after the decimal?

So, if it is max significant digits, then 123.45678901, a valid MINA quantity, will be truncated to 123.456789.

Reply:
That's correct. I've pushed a commit to fix this.